### PR TITLE
[vlm] feat: add MiniCPM-V-4.6 support

### PR DESCRIPTION
**Pull request by Zachary-wW**
## Summary

This PR adds end-to-end LoongForge support for `openbmb/MiniCPM-V-4.6`, including:

- the Qwen3.5-based hybrid gated-delta/full-attention language decoder;
- the MiniCPM-V vision tower and layer-6 ViT merger;
- MiniCPM-V image preprocessing, chat templating, and multimodal task encoding;
- full-model SFT and pretraining examples;
- bidirectional Hugging Face <-> Megatron Core checkpoint conversion;
- model-specific handling for Transformer Engine `_extra_state` compatibility;
- focused model, data, LoRA, and checkpoint-conversion tests.

The implementation keeps the existing Qwen3.5 foundation and generic checkpoint converters unchanged. MiniCPM-specific behavior is confined to the MiniCPM model, configuration, data, and example files.

## Checkpoint Conversion

- Converts the released HF checkpoint into MCore and merges language, vision, ViT patch, and merger weights.
- Converts the merged MCore checkpoint back to HF.
- Preserves the MiniCPM ViT and merger parameter names and tensor layouts.
- Handles TE `_extra_state` compatibility through MiniCPM model load hooks; no MiniCPM-specific converter scripts are required.

The released checkpoint declares MTP support but contains no MTP tensor keys. The conversion path therefore does not fabricate source MTP weights; MTP remains available as a runtime configuration.

## Swift Alignment

The fixed comparison uses the same model configuration, weights, input batch, dtype, seed, optimizer, scheduler, and attention backend as the reference MS-Swift setup. MTP is disabled for the comparison.

The following chart compares the 10 optimizer steps (the first line is MS-Swift; the second line is LoongForge):

| Step | MS-Swift | LoongForge | Absolute difference |
| ---: | ---: | ---: | ---: |
| 1 | 6.559576511 | 6.559576700 | 0.000000189 |
| 2 | 3.911621809 | 3.911621803 | 0.000000006 |
| 3 | 2.305612803 | 2.301189866 | 0.004422936 |
| 4 | 1.710727811 | 1.707872967 | 0.002854843 |
| 5 | 0.716023266 | 0.717803245 | 0.001779979 |
| 6 | 0.355155975 | 0.358439689 | 0.003283715 |
| 7 | 0.140928328 | 0.140765501 | 0.000162827 |
| 8 | 0.031386301 | 0.027902570 | 0.003483731 |
| 9 | 0.040537037 | 0.045103772 | 0.004566735 |
| 10 | 0.011375484 | 0.010368158 | 0.001007326 |

## Validation

- [x] Strictly load all 779 tensors from the released HF checkpoint.
- [x] Verify HF -> MCore -> HF tensor roundtrip.
- [x] Compare 10 optimizer steps with MS-Swift.
- [x] Run the requested eight-GPU parallel topologies.
- [x] Run a DP8 sequence-packing smoke test with MTP enabled.
- [x] Run the focused LoongForge tests.
- [x] Run checkpoint-converter configuration, Python syntax, and shell syntax checks.

The three CUDA/attention package pins are intentionally not added to this PR. They will be managed by the repository-wide CI dependency update.

### CODE REVIEW

**Comment by Zachary-wW** [comment]
@claude review this PR

File: loongforge/data/chat_template.py [hunk 1]
```diff
@@ -54,6 +55,25 @@
 SlotsType = Sequence[Union[str, Set[str], Dict[str, str]]]
 
 
+class _LazyMMPlugin:
```
Comment (nullnonenilNULL) [inline loongforge/data/chat_template.py]: 延迟加载在这里的意义是什么？

File: loongforge/data/chat_template.py [hunk 2]
```diff
@@ -953,6 +973,16 @@ def _read_builtin_chat_template(filename: str) -> str:
     chat_template=_read_builtin_chat_template("qwen_chat_hf_training.jinja"),
 )
 
+_register_chat_template(
+    name="minicpm-v-4.6-hf",
+    cls=HFChatTemplate,
+    chat_template=_read_builtin_chat_template("minicpm_v_4_6_hf_training.jinja"),
+    mm_plugin=_LazyMMPlugin(
```
Comment (nullnonenilNULL) [inline loongforge/data/chat_template.py]: 其他的模型为什么不需要 lazymmplugin？或者是否需要改成延迟加载

File: loongforge/train/training_utils.py [hunk 3]
```diff
@@ -836,11 +837,31 @@ def peft_pre_wrap_hook(args, model, peft_class):
                 ignore_ckpt_step=True,  # ckpt_step applies only to adapter checkpoints, not pretrained base model
             )
 
+        custom_peft_class = getattr(model_config, "peft_class", None)
         if "VLM" in type(model_config.peft_config).__name__:
             peft_config = check_vlm_peft_config(model_config)
-            peft_class = VLMLoRA(**asdict(peft_config))
+            peft_kwargs = asdict(peft_config)
```
Comment (nullnonenilNULL) [inline loongforge/train/training_utils.py:843]: 已有的 lora 有什么问题导致需要自定义

File: loongforge/data/chat_template.py (hunk 2 above)
Comment (Zachary-wW) [reply loongforge/data/chat_template.py]: 已移除 _LazyMMPlugin，改为直接注册 MiniCPMV46Plugin。之前延迟加载的目的是在避免全局 chat-template 注册时加载 torchvision 或图像处理器

File: loongforge/train/training_utils.py (hunk 3 above)
Comment (Zachary-wW) [reply loongforge/train/training_utils.py]: - MiniCPM-V 4.6 包含 Qwen3.5 风格的 in_proj_qkvz 和 in_proj_ba。前向计算时，会按照固定的组件边界和顺序拆分这些 packed projection（e.g. in_proj_qkvz 拆成 qkv 和 z，in_proj_ba 拆成 b 和 a），然后分别执行 F.linear，普通 LoRA wrapper 无法表达各 component 的 split size、排列顺序和 adapter 注入位置
- 这条路径直接调用拆分后的 F.linear，不会调用包裹整个 fused module 的普通 LoRA forward(), 自定义 LoRA 的作用是为每个拆分后的 component 分别注入 adapter，同时保持原有 LoRA 逻辑用于普通 projection。
